providers/fips/fipsprov.c

```diff
@@ -683,6 +709,39 @@ static const OSSL_ALGORITHM *fips_query(void *provctx, int operation_id,
     return NULL;
 }
 
+static const OSSL_ALGORITHM *fips_query_internal(void *provctx, int operation_id,
+                                                 int *no_cache)
+{
+    *no_cache = 0;
+
+    if (!ossl_prov_is_running())
+        return NULL;
+
+    switch (operation_id) {
+    case OSSL_OP_DIGEST:
+        return fips_digests;
+    case OSSL_OP_CIPHER:
+        return exported_fips_ciphers;
+    case OSSL_OP_MAC:
+        return fips_macs_internal;
```

Comment:
Given that this function is identical to `fips_query` with the exception of this one line, would it be better have one call the other - with this function just handling `OSSL_OP_MAC` independently? I worry that future updates will cause these two functions to get out of sync with each other.

Reply:
The relatively small number of algorithms that are internally fetched by the FIPS provider would have allowed a simplified of this but reusing the external code is a reasonable solution too.